src/components/task/repo/task.repo.js

```diff
@@ -0,0 +1,133 @@
+const _ = require('lodash');
+const { SubmissionModel, HomeworkModel } = require('../db');
+
+const { updateManyResult, deleteManyResult } = require('../../helper/repo.helper');
+
+const { isValid: isValidObjectId } = require('../../../helper/compare').ObjectId;
+const { missingParameters } = require('../../../errors/helper/assertionErrorHelper');
+const { AssertionError } = require('../../../errors');
+
+const validateObjectId = (objectId) => {
+	const missingProps = {};
+	for (const prop in objectId) {
+		if (!isValidObjectId(objectId[prop])) {
+			missingProps[prop] = objectId[prop];
+		}
+	}
+	if (!_.isEmpty(missingProps)) {
+		throw new AssertionError(missingParameters({ ...missingProps }));
+	}
+};
+
+/** Homeworks */
+const privateHomeworkQuery = (userId) => ({ private: true, teacherId: userId });
+const publicHomeworkQuery = (userId) => ({ private: false, teacherId: userId });
+const archivedHomeworkQuery = (userId) => ({ archived: userId });
+
+/**
+ * @param {BSON|BsonString} userId
+ * @return {Array}
+ */
+const findPrivateHomeworksByUser = async (userId) => {
+	const result = await HomeworkModel.find(privateHomeworkQuery(userId)).lean().exec();
```

Comment:
i would add validating the userId here and in findPublicHomeworkIdsByUser as the intention here is to have it given